Lib/glyphsLib/builder/builders.py

```diff
@@ -411,6 +412,18 @@ def instance_data(self):
     )
 
 
+def preflight_glyphs(font):
+    # Run a set of transformations over a GSFont object to make
+    # it easier to convert to UFO; resolve all the "smart stuff".
+
+    # We could "font = copy.deepcopy(font)" here, since we'll be
+    # modifying the original. But that's a pain and we won't do it
+    # until someone complains.
```

Comment:
I wonder if we were modifying the input GSFont previously, before this PR; I'd expect we should not, though I can see how this might be convenient for you here..
How about you document that this preflight works in place (so if one cares, they can pass in a copy) and instead of doing it automatically inside the UFOBuilder constructor, you run it immediately before the latter is run?
Feels like the two should be kept distinct from one another instead of conflated like this.

Reply:
I undestand, but it still feels strange to me that we now suddenly choose to modify the input GSFont in place just because it's handy, we can't be 100% sure that one (ourselves even in the unit tests) aren't calling the existing to_ufos or to_designspace or UFOBuilder methods on the _same_ GSFont object with different parameters maybe or after having changed something in it (with the assumption that these methods weren't actually supposed to have side effects but just return something new each time and keep the input unchanged).